components/server/src/workspace/workspace-service.ts

```diff
@@ -191,7 +192,15 @@ export class WorkspaceService {
             );
             throw err;
         }
-        await this.updateDeletionEligabilityTime(user.id, workspace.id);
+        const update = this.updateDeletionEligabilityTime(user.id, workspace.id);
+        if (WithPrebuild.is(workspace.context) && workspace.context.prebuildWorkspaceId) {
```

Comment:
@svenefftinge Can we separate theses two more, somehow? This code is quite confusing, and mixes two aspects (update this workspaces, vs. update some other prebuiltWorkspace).
Why not:
```
await this.updateDeletionEligabilityTime(user.id, workspace.id);

(async () => { ... })().catch(err => ...);
```
That would read less confusing to me. Bonus points for naming the function. :smile:

Reply:
I don't understand what you mean 🤷 Can you make your proposal more detailed, the code is only showing some unknown immediate execution of a function literal ?

Reply:
Plus it's not on the hot path and does not fail the start of the workspace.

Reply:
Leaving it up to you whether you want to adjust it or not :+1:

Reply:
Ok, a bit of a matter of test. Readybility is imho not better with the lambda construct. but we could consider if we need to await any of the calls to `updateDeletionEligabilityTime` at all. As they only update the eligibility time and it doesn't need to be kept in sync. But I would do this for all calls not just for the prebuild case.

Reply:
Thinking about it, maybe this is actually a good follow up. Adding additional DB calls to the start workspace path should only await if really necessary.